### PR TITLE
Fix streaming ci failure

### LINKED ISSUE #12834: Remove load_code_from_local
### Describe your feature request
Currently, the python code can be loaded from gcs or local, but can't can loaded from both side. It would be great if we can try load code from local first, and load from gcs if we can't load cpode from local.

### PR DESCRIPTION
**Pull request by chaokunyang**
## Why are these changes needed?
This PR fix streaming failure by doing following things:
* fix load_code_from_local not set in driver which cause streamning ci failure:
```
python/ray/streaming/context.py:48: in build
    ctx = StreamingContext()
python/ray/streaming/context.py:53: in __init__
    self.__gateway_client = GatewayClient()
python/ray/streaming/runtime/gateway_client.py:17: in __init__
    GatewayClient._PYTHON_GATEWAY_CLASSNAME).remote()
python/ray/actor.py:408: in remote
    return self._remote(args=args, kwargs=kwargs)
python/ray/actor.py:655: in _remote
    creation_args = cross_language.format_args(worker, args, kwargs)
```

* the new scheduler doesn't support resource which cause streaming workers can't be created, use old scheduler to workaround  by set `RAY_ENABLE_NEW_SCHEDULER=0`. We'll use placementgroup later for scheduling. 
* add exception when jobmaster can't create workers.
* fix channel id length
## Related issue number
#12834
#13021

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by chaokunyang** [comment]
I'll remove all load_code_from_local flag if trying load form lcoal then gcs is acceptable.

File: streaming/python/tests/test_hybrid_stream.py [hunk 1]
```diff
@@ -31,12 +31,9 @@ def test_hybrid_stream():
         "../../../bazel-bin/streaming/java/all_streaming_tests_deploy.jar")
     jar_path = os.path.abspath(jar_path)
     print("jar_path", jar_path)
-    java_worker_options = json.dumps(["-classpath", jar_path])
-    print("java_worker_options", java_worker_options)
     assert not ray.is_initialized()
     ray.init(
-        job_config=ray.job_config.JobConfig(code_search_path=sys.path),
-        _java_worker_options=java_worker_options)
+        job_config=ray.job_config.JobConfig(code_search_path=[jar_path]))
```
Comment (kfstorm) [inline streaming/python/tests/test_hybrid_stream.py:36]: Please update other streaming test cases with `code_search_path` configured as well. I believe we chan change them from `ray.init(job_config=ray.job_config.JobConfig(code_search_path=sys.path))` to `ray.init()`.

File: python/ray/function_manager.py [hunk 2]
```diff
@@ -230,14 +230,14 @@ def get_execution_info(self, job_id, function_descriptor):
...
-        if self._worker.load_code_from_local:
+        try:
             # Load function from local code.
             # Currently, we don't support isolating code by jobs,
             # thus always set job ID to NIL here.
             job_id = ray.JobID.nil()
             if not function_descriptor.is_actor_method():
                 self._load_function_from_local(job_id, function_descriptor)
-        else:
+        except RuntimeError as e:
```
Comment (kfstorm) [inline python/ray/function_manager.py:240]: What's the performance impact for users who want to load code from GCS (which is the majority)?

File: python/ray/cross_language.py [hunk 3]
```diff
@@ -22,9 +22,6 @@ def format_args(worker, args, kwargs):
     Returns:
         List of args and kwargs (if supported).
     """
-    if not worker.load_code_from_local:
```
Comment (kfstorm) [inline python/ray/cross_language.py:25]: Would be better if you could delete all occurrences of `load_code_from_local` in the code base.

**Comment by kfstorm** [comment]
I'm not sure why this PR is the correct way to fix the streaming test case. According to https://github.com/ray-project/ray/pull/10573/files#diff-b3d68c2dfbecb41ea9d43d58f38f13fe9042e041d04e86f296b4d9a7725e8471, `local_code_from_local` will be set to true automatically if the code search path is set.

File: streaming/python/tests/test_hybrid_stream.py (hunk 1 above)
Comment (lixin-wei) [inline streaming/python/tests/test_hybrid_stream.py:36]: Please modify other streaming tests as well, check the CI for failed tests.

**Review by lixin-wei — DISMISSED** [review]

**Comment by chaokunyang** [comment]
> I'm not sure if this is the right way to fix the streaming test case. According to https://github.com/ray-project/ray/pull/10573/files#diff-b3d68c2dfbecb41ea9d43d58f38f13fe9042e041d04e86f296b4d9a7725e8471, `local_code_from_local` will be set to true automatically if the code search path is set.

That PR didn't set load_code_from_local for driver if code search path is not null

**Comment by kfstorm** [comment]
@chaokunyang I mean the logic is already in master. It was added [here](https://github.com/ray-project/ray/pull/10573/files#diff-b3d68c2dfbecb41ea9d43d58f38f13fe9042e041d04e86f296b4d9a7725e8471)

**Comment by kfstorm** [comment]
@chaokunyang What's the case that a driver needs `load_code_from_local` to be true? A driver should never execute tasks.

**Comment by chaokunyang** [comment]
> @chaokunyang What's the case that a driver needs `load_code_from_local` to be true? A driver should never execute tasks.

If the driver need to create a java actor or make a java task call from the python driver, the driver need the load_code_from local to be set, or else the `cross_language.format_args` will throws an error.

**Comment by chaokunyang** [comment]
After offline discusstion with @fyrestone , he will submit a more shophicated solution on code load from local or gcs. I'll revert the load_code_from_local and set load_code_from_local flag in driver

File: python/ray/function_manager.py (hunk 2 above)
Comment (chaokunyang) [reply python/ray/function_manager.py]: Let's remove it in another pr

File: python/ray/cross_language.py (hunk 3 above)
Comment (chaokunyang) [reply python/ray/cross_language.py]: Let's remove it in another pr

**Comment by kfstorm** [comment]
@chaokunyang Streaming tests failed.

**Comment by ericl** [comment]
Tests still seem to be failing

**Comment by chaokunyang** [comment]
@ericl All ci tests pass now.

**Review by raulchen — APPROVED** [review]

**Pull request merged**